File: example/src/main/java/ru/virgil/example/order/BuyingOrder.java
package ru.virgil.example.order;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import ru.virgil.example.system.IdentifiedEntity;
import ru.virgil.example.truck.Truck;
import ru.virgil.example.user.UserDetails;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
public class BuyingOrder extends IdentifiedEntity {

    @ManyToOne
    private UserDetails owner;
    @OneToMany
    private List<Truck> trucks = new ArrayList<>();
    private String description;

}

File: example/src/main/java/ru/virgil/example/order/BuyingOrder.kt
package ru.virgil.example.order

import ru.virgil.example.system.entity.OwnedEntity
import ru.virgil.example.truck.Truck
import ru.virgil.example.user.UserDetails
import javax.persistence.Entity
import javax.persistence.ManyToOne

/**
 * Это фикс поведения JPA для двусторонних связей. Проблема тянется еще с JPA 2,
 * но ее не фиксят, чтобы не сломать совместимость. Читал на StackOverflow
 */
private const val CONNECTION = "buying_order"

@Entity
class BuyingOrder(
    owner: UserDetails,
    @ManyToOne
    var truck: Truck,
    var description: String?,
) : OwnedEntity(owner)
